metaflow/decorators.py

```diff
@@ -257,6 +281,34 @@ class MyDecorator(StepDecorator):
                   pass them around with every lifecycle call.
     """
 
+    cli_options = {}
+    # `cli_options` is similar to the one the flow decorator. It will be used to
+    # pass options to the step decorator from the cli.
+
+    @classmethod
+    def step_options_init(cls, flow, options_dict):
```

Comment:
This can probably be combined with the config thing (ie: we probably don't need to add two new hooks)